LosAngelesDataProject/LADP
# User-Register-Page-II

▶️ Task
https://trello.com/c/stWZMO2f/21-user-register-page-ii

📓 Description
Implemented client-side validation for the registration form to ensure all required fields are correctly filled out before submission, using Formik and Yup or similar tools. Integrated a React Wizard for the registration process. Embedded the user registration form within a modal in the register component. Ensured the functionality of the finish button. Added success, error, and loading handlers. Organized all registration components into a dedicated directory. Reused the header component for consistency. Created a modal to host the registration wizards.

📸 UI
![Screenshot 2024-06-07 171128](https://github.com/LosAngelesDataProject/LADP/assets/151715437/ba26bd7c-ef17-46f6-8947-a62eca54225a)
![Screenshot 2024-06-07 171203](https://github.com/LosAngelesDataProject/LADP/assets/151715437/b668f39c-9564-44a1-92d1-78be228401ef)
![Screenshot 2024-06-07 172551](https://github.com/LosAngelesDataProject/LADP/assets/151715437/a6d0fbc4-5d6d-487f-8c38-d897029eaa6b)
![Screenshot 2024-06-07 171216](https://github.com/LosAngelesDataProject/LADP/assets/151715437/f07a58b2-3738-4f5b-a22b-adcbc909aed2)
![Screenshot 2024-06-07 171233](https://github.com/LosAngelesDataProject/LADP/assets/151715437/37f7928c-a234-4410-bd81-79c9f5f306b3)
![Screenshot 2024-06-07 171315](https://github.com/LosAngelesDataProject/LADP/assets/151715437/95051913-a836-4e2c-9df0-9491be0d137b)
![Screenshot 2024-06-07 171342](https://github.com/LosAngelesDataProject/LADP/assets/151715437/ea35d1fa-b57f-4ca1-8395-5b00ff461348)


📁 SQL
N/A

📶 Third Party API
Utilized Formik, Yup, Bootstrap, React Wizard, Toatsr, React Icons, Bootstrap-Icons

🧪 How to test
How can a peer test this work ?

Comments
You will notice a TwoFactorAuth component that is not currently used for registration. I thought it would be more appropriate for the login page, so I left the file in place for future use if needed.
